### PR TITLE
NEP: Improve Value Based Casting paragraph in NEP 40

### PR DESCRIPTION
**Pull request by anirudh2290**
nit: The fact that ``5`` was referring to the expression above was unclear to me and I added a few words for that in the doc.

### CODE REVIEW

File: doc/neps/nep-0040-legacy-datatype-impl.rst [hunk 1]
```diff
@@ -111,9 +111,9 @@ represented as an ``int8`` or ``int16`` (the smallest possible data type).
 
 NumPy currently does this even for NumPy scalars and zero dimensional arrays,
```
Comment (rossbar) [inline doc/neps/nep-0040-legacy-datatype-impl.rst:112]:
```suggestion
NumPy currently does this even for NumPy scalars and zero-dimensional arrays,
```

File: doc/neps/nep-0040-legacy-datatype-impl.rst [hunk 2]
```diff
@@ -111,9 +111,9 @@ represented as an ``int8`` or ``int16`` (the smallest possible data type).
...
 so that replacing ``5`` with ``np.int64(5)`` or ``np.array(5, dtype="int64")``
-will lead to the same results, and thus ignores the existing datatype.
-The same logic also applies to floating point scalars, which are allowed to
-lose precision.
+in the above expression will lead to the same results, and thus ignores the
+existing datatype. The same logic also applies to floating point scalars,
```
Comment (rossbar) [inline doc/neps/nep-0040-legacy-datatype-impl.rst:115]:
```suggestion
existing datatype. The same logic also applies to floating-point scalars,
```

**Review by rossbar — APPROVED** [review]
IMO the proposed addition does indeed improve clarity.

Added a few grammatical nits since the text is being reviewed - please ignore if they stand in the way of the original proposed changes.

**Comment by anirudh2290** [comment]
Thank you @rossbar for your review ! I have applied your proposed changes.

**Comment by seberg** [comment]
Thanks!

**Pull request merged**